include/criteria.h

```diff
@@ -0,0 +1,45 @@
+#ifndef _MAKO_CRITERIA_H
+#define _MAKO_CRITERIA_H
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "notification.h"
+
+struct mako_config;
+
+// Stores whether or not each field was part of the criteria specification, so
+// that, for example, "not actionable" can be distinguished from "don't care".
+// This is unnecessary for string fields, but it's best to just keep it
+// consistent.
+struct mako_criteria_spec {
```

Comment:
Instead, we could have an `enum mako_criteria_bool` with three variants: none, true, false.

Reply:
I had that originally, but then immediately ran into needing something for `urgency` as well.

Reply:
Hmm, I see. We could then store bools alongside the other fields, but this is fine too.

Reply:
I liked having the symmetry in `match_criteria`, but if it becomes too annoying to maintain these duplicate definitions, it should be easy enough to flatten.